src/import_pipeline.cc

```diff
@@ -216,6 +224,18 @@ void QueryDb_OnIndexed(QueueManager* queue,
                        SemanticHighlightSymbolCache* semantic_cache,
                        WorkingFiles* working_files,
                        Index_OnIndexed* response) {
+
+  if (response->update.file_id < 0) { // dummy
```

Comment:
Semantic highlighting does work reliably when the project is still loading is because `message_handler.cc` uses:

```c++
    switch (sym.kind) {
      case SymbolKind::Func: {
        const QueryFunc& func = db->GetFunc(sym);
        const QueryFunc::Def* def = func.AnyDef(); ///////// definition required, declaration is ignored
        if (!def)
          continue;  // applies to for loop
```

I think this patch is very promising to improve the situation! I occasionally observe that some files are not highlighted because I am unable to narrow it down (does not look like a forward-line out-of-range error)

I'll try this patch tomorrow on LLVM